File: pp-backend/src/main/java/com/ppbackend/user/repository/UserRepository.java
package com.ppbackend.user.repository;

import com.ppbackend.user.model.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends ElasticsearchRepository<User, String> {
    User findByUsername(String username);
    User findByUuid(String Uuid);
    Page<User> findAll(Pageable pageable);
}

File: src/main/kotlin/com/picuser/repository/UserRepository.kt
package com.picuser.repository

import com.picuser.entities.User
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.elasticsearch.annotations.Query
import org.springframework.data.elasticsearch.repository.ElasticsearchRepository

interface UserRepository : ElasticsearchRepository<User, String> {

    @Query("{\"bool\":{\"must\":[{\"match\":{\"name\":\"?0\"}}]}}")
    fun findUser(text: String, pageable: Pageable): Page<User>
}